src/ng/compile.js

```diff
@@ -3350,9 +3459,56 @@ function $CompileProvider($provide, $$sanitizeUriProvider) {
       }
     }
 
+    function getTrustedPropContext(nodeName, propNormalizedName) {
+      var prop = propNormalizedName.toLowerCase();
+      return PROP_CONTEXTS[nodeName + "|" + prop] || PROP_CONTEXTS["*|" + prop];
+    }
+
+    function sanitizeSrcsetPropertyValue(value) {
+      return sanitizeSrcset($sce.valueOf(value), 'ng-prop-srcset');
+    }
+    function addPropertyDirective(node, directives, attrName, propName) {
+      if (EVENT_HANDLER_ATTR_REGEXP.test(propName)) {
+        throw $compileMinErr('nodomevents',
+            'Property bindings for HTML DOM event properties are disallowed.  Please use the ' +
+                'ng- versions (such as ng-click or ng-on-click instead of ng-prop-onclick) instead.');
+      }
+
+      var nodeName = nodeName_(node);
+      var trustedContext = getTrustedPropContext(nodeName, propName);
+
+      var sanitizer = identity;
+      // Sanitize img[srcset] + source[srcset] values.
+      if ((nodeName === 'img' || nodeName === 'source') && propName === 'srcset') {
+        sanitizer = sanitizeSrcsetPropertyValue;
+      } else if (trustedContext) {
+        sanitizer = $sce.getTrusted.bind($sce, trustedContext);
+      }
+
+      directives.push({
+        priority: 100,
+        compile: function ngPropCompileFn(_, attr) {
+          var fn = $parse(attr[attrName]);
+          return {
+            pre: function ngPropPreLinkFn(scope, $element) {
+              scope.$watch(fn, function propertyWatchActionFn(value) {
```

Comment:
Do we need to look at the implementation of `ngBindHtml` here for the watcher, where the inner value is what is watched rather than the trusted object; and also consider calls to `$$bindingInfo` and friends.

See https://github.com/angular/angular.js/blob/74726b4f7934720c8106dfb7212b868ad5efc111/src/ng/directive/ngBind.js#L189-L206

Reply:
I've essentially copied `ngBindHtml`: https://github.com/angular/angular.js/pull/16614/commits/f547b0da82724b1d9d2f3ec38f8c7332bd4f1933